web/pdf_viewer.js

```diff
@@ -1992,43 +1992,63 @@ class PDFViewer {
 
   /**
    * Increase the current zoom level one, or more, times.
-   * @param {number} [steps] - Defaults to zooming once.
    * @param {Object|null} [options]
    */
-  increaseScale(steps = 1, options = null) {
+  increaseScale(options = null) {
     if (!this.pdfDocument) {
       return;
     }
-    let newScale = this._currentScale;
-    do {
-      newScale = (newScale * DEFAULT_SCALE_DELTA).toFixed(2);
-      newScale = Math.ceil(newScale * 10) / 10;
-      newScale = Math.min(MAX_SCALE, newScale);
-    } while (--steps > 0 && newScale < MAX_SCALE);
 
     options ||= Object.create(null);
-    options.noScroll = false;
+
+    let newScale;
+    if (options.scaleFactor) {
+      newScale = Math.min(
+        MAX_SCALE,
+        Math.round(this._currentScale * options.scaleFactor * 100) / 100
+      );
+    } else {
+      let steps = options.steps ?? 1;
+      newScale = this._currentScale;
+      do {
+        newScale = (newScale * DEFAULT_SCALE_DELTA).toFixed(2);
+        newScale = Math.ceil(newScale * 10) / 10;
+        newScale = Math.min(MAX_SCALE, newScale);
+      } while (--steps > 0 && newScale < MAX_SCALE);
+
+      options.noScroll = false;
+    }
     this._setScale(newScale, options);
   }
 
   /**
    * Decrease the current zoom level one, or more, times.
-   * @param {number} [steps] - Defaults to zooming once.
    * @param {Object|null} [options]
    */
-  decreaseScale(steps = 1, options = null) {
+  decreaseScale(options = null) {
     if (!this.pdfDocument) {
       return;
     }
-    let newScale = this._currentScale;
-    do {
-      newScale = (newScale / DEFAULT_SCALE_DELTA).toFixed(2);
-      newScale = Math.floor(newScale * 10) / 10;
-      newScale = Math.max(MIN_SCALE, newScale);
-    } while (--steps > 0 && newScale > MIN_SCALE);
 
     options ||= Object.create(null);
-    options.noScroll = false;
+
+    let newScale;
+    if (options.scaleFactor) {
```

Comment:
Not enough validation here, this needs to be
```suggestion
    if (options.scaleFactor > 0 && options.scaleFactor < 1) {
```